bin/dry-run.rb

```diff
@@ -592,6 +592,8 @@ def security_advisories
 end
 
 def peer_dependencies_can_update?(checker, reqs_to_unlock)
+  return false if $options[:security_updates_only]
```

Comment:
This makes sense on a third take.

Initially I was thinking you could get duplicate PRs if a security update also needs to update a peer dependency. If another update tries to update the peer dep you would end up with two PRs that have an identical diff but the title/body would say PR 1 updates: A + B and PR 2 updates B + A.

I think my initial assumption is incorrect as you only get into the identical diff case if you're doing version updates, trying to update every dependency in turn, whereas security updates are only ever trying to update one dependency.

There might be a slightly odd case if two security advisories exist, one for a transitive dep and one for a top-level dep that needs to be unlocked. In this case you would not get an identical diff, but the two PRs would conflict as they would both update the top level dep but this doesn't seem worth handing yet..

I was wondering if it would be worth documenting some of this weirdness here, in case someone runs into this case and comes along looking for explanations in future?

Reply:
@feelepxyz I've tried to improve the docs around that call in https://github.com/dependabot/dependabot-core/pull/5239/commits/a70b5435889ead31f8de739319e76d2c6945b9b0. Does that sound accurate to you?

I also spotted a small optimization in https://github.com/dependabot/dependabot-core/pull/5239/commits/2b2cc38665b037dd07e9b18d492d85a2789595bc. Is there any reason we'd need to make a 2nd call to `checker.updated_dependencies(requirements_to_unlock: reqs_to_unlock)`?